projects/angular/src/data/datagrid/datagrid-pagination.spec.ts

```diff
@@ -256,7 +256,7 @@ export default function (): void {
         expect(context.testComponent.current.toString()).toBe('4');
       });
 
-      it('changes the current page on blur', function () {
+      it('does not change the current page on blur', function () {
```

Comment:
This seems like it would be considered a breaking change

Reply:
It's a functionality logged as a bug. If we accept the previous one was erroneous, then it's a bug fix.
It's not API of any kind that users can build functionality upon.
Same way, if a color does not meet contrast, we change it. And though it "could" be a visual breaking change, we consider it a bug fix.

Reply:
We have a unit test for very specific behavior that was changed to do the complete opposite. That feels to me like a breaking behavioral change, not a visual one. 

If there wasn't a unit test for it, I wouldn't have thought twice about it, but why do we have a unit test that tests specifically for the bug behavior?

Reply:
I agree with Ashley.

Reply:
Also, I don't quite understand the bug. What are the steps to reproduce?

Reply:
Actually, sometimes the tests are just plain wrong. See my fix in #317. Although this one changes tested behavior and mine just fixes an incorrect expected value.

Reply:
Steps to reproduce:
1. Open a datagrid with pagination in it.
2. Tab to the current page input in the footer (or click on it). 
3. Type a new page number (do not hit Enter).
4. Tab out of the input field.
5. Observe the current datagrid page changes.

This is the bug. According to A11Y, an action can not occur without active confirmation. The best solution would be to have a button next to the input, but hitting Enter can be accepted as good enough alternative, as a button will visually overload the component, which already has enough buttons.
Change of the page, happening on focus out of the input should not occur though.

CC: @AmyLiNow 

I don't quite understand the comment about the wrong tests and the provided example.
The tests here used to verify the old behavior. They were correct once this was the expected behavior.
Once it's found that this behavior needs to be changed, we also need to change the tests.

Reply:
> ...but why do we have a unit test that tests specifically for the behavior of the bug?
Code coverage?

Reply:
I don't feel convinced that this specific change needs major release.
Though it's a good food for our old "what is a breaking change" talk.

Reply:
> The tests here used to verify the old behavior. They were correct once this was the expected behavior.
Once it's found that this behavior needs to be changed, we also need to change the tests.

Sure, the behavior needs to change because of A11Y, but that's still a _breaking behavioral change_. That was the explicit and intended behavior before - we even had unit tests for it. Now we're changing it to do the complete opposite - that's breaking.

If we didn't have the unit tests before, then I could believe the old behavior was just a bug and could be changed in a minor/patch release. But the fact that we did have unit tests for it means that the old behavior was intentional (even if ultimately wrong) and therefore cannot be changed until the next major version.